py/server/deephaven/updateby.py

```diff
@@ -759,4 +997,254 @@ def rolling_prod_time(ts_col: str, cols: Union[str, List[str]], rev_time: Union[
         fwd_time = _JDateTimeUtils.expressionToNanos(fwd_time) if isinstance(fwd_time, str) else fwd_time
         return UpdateByOperation(j_updateby_op=_JUpdateByOperation.RollingProduct(ts_col, rev_time, fwd_time, *cols))
     except Exception as e:
-        raise DHError(e, "failed to create a rolling product (time) UpdateByOperation.") from e
\ No newline at end of file
+        raise DHError(e, "failed to create a rolling product (time) UpdateByOperation.") from e
+
+
+def rolling_count_tick(cols: Union[str, List[str]], rev_ticks: int, fwd_ticks: int = 0) -> UpdateByOperation:
+    """Creates a rolling count UpdateByOperation for the supplied column names, using ticks as the windowing unit. Ticks
+    are row counts, and you may specify the reverse and forward window in number of rows to include. The current row
+    is considered to belong to the reverse window but not the forward window. Also, negative values are allowed and
+    can be used to generate completely forward or completely reverse windows.
+
+    Here are some examples of window values:
+        rev_ticks = 1, fwd_ticks = 0 - contains only the current row
+        rev_ticks = 10, fwd_ticks = 0 - contains 9 previous rows and the current row
+        rev_ticks = 0, fwd_ticks = 10 - contains the following 10 rows, excludes the current row
+        rev_ticks = 10, fwd_ticks = 10 - contains the previous 9 rows, the current row and the 10 rows following
+        rev_ticks = 10, fwd_ticks = -5 - contains 5 rows, beginning at 9 rows before, ending at 5 rows before the
+            current row (inclusive)
+        rev_ticks = 11, fwd_ticks = -1 - contains 10 rows, beginning at 10 rows before, ending at 1 row before the
+            current row (inclusive)
+        rev_ticks = -5, fwd_ticks = 10 - contains 5 rows, beginning 5 rows following, ending at 10 rows following the
+            current row (inclusive)
+
+
+    Args:
+        cols (Union[str, List[str]]): the column(s) to be operated on, can include expressions to rename the output,
+            i.e. "new_col = col"; when empty, update_by perform the rolling count operation on all columns.
+        rev_ticks (int): the look-behind window size (in rows/ticks)
+        fwd_ticks (int): the look-forward window size (int rows/ticks), default is 0
+
+    Returns:
+        an UpdateByOperation
+
+    Raises:
+        DHError
+    """
+    try:
+        cols = to_sequence(cols)
+        return UpdateByOperation(j_updateby_op=_JUpdateByOperation.RollingCount(rev_ticks, fwd_ticks, *cols))
+    except Exception as e:
+        raise DHError(e, "failed to create a rolling count (tick) UpdateByOperation.") from e
+
+def rolling_count_time(ts_col: str, cols: Union[str, List[str]], rev_time: Union[int, str],
```

Comment:
Looking at this, I'm noticing a naming difference that I want to point out before we have DnD users.  Here we have `rolling_count_tick` and `rolling_count_time` but earlier we had `ema_tick_decay` and `ema_time_decay` (or whatever it was).  Should we include the `_decay` in the other methods?

I'm not sure what my thoughts are, but I'm pointing it out before we get locked in forever.
@lbooker42 @jmao-denver

Reply:
I dislike adding `decay` to Rolling operators; they do not decay like exponential ops. Actually I prefer to have the em function as `ema_ticks`/`ema_time`/`ems_ticks` etc. without `decay` attached.  Any user of these functions should already understand the decay mechanism built in.

Reply:
I agree with @lbooker42, I am for removing 'decay' from the function names.

Reply:
I was suggesting removing `_decay` for consistency -- not adding it to everything.  Shorter is better as long as we are clear.

Reply:
I see that my initial writeup was garbage, and I did say "include".